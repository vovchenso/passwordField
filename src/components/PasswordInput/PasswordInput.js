import React from 'react';
import {
    StyleSheet,
    View,
    TextInput,
    Text,
    TouchableOpacity
} from 'react-native';

export default class PasswordInput extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            show: false,
            focus: false
        };
    }

    onShowHide = () => {
        this.setState({
            show: !this.state.show
        });
    };

    onFocusBlur = () => {
        this.setState({
            focus: !this.state.focus
        });
    };

    render() {
        const { placeholder, onChange } = this.props;
        const { show, focus } = this.state;

        const inputStyles = [
            styles.input,
            focus && styles.inputActive,
            show && styles.inputShow
        ];

        const textStyle = [
            styles.buttonText,
            show ? styles.buttonTextShow : styles.buttonTextHide
        ];

        return (
            <View style={styles.wrapper}>
                <TextInput
                    onFocus={ this.onFocusBlur }
                    onBlur={ this.onFocusBlur }
                    style={ inputStyles }
                    placeholder={ placeholder }
                    secureTextEntry={ !show }
                    onChangeText={ value => onChange && onChange(value) }
                />
                <TouchableOpacity
                    accessibilityTraits="button"
                    accessibilityComponentType="button"
                    onPress={ this.onShowHide }
                    style={ styles.button }
                >
                    <Text style={ textStyle }>
                        { (show ? 'HIDE' : 'SHOW') }
                    </Text>
                </TouchableOpacity>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    input: {
        height: 40,
        borderColor: 'gray',
        borderWidth: 1,
        borderRadius: 5,
        padding: 5,
        color: '#a0a0a0'
    },
    inputActive: {
        borderColor: '#88f',
    },
    inputShow: {
        borderColor: '#f22',
        backgroundColor: '#fbb',
        color: '#000'
    },
    button: {
        position: 'absolute',
        right: 0,
        bottom: 0,
        top: 0,
        flex: 1,
        justifyContent: 'center',
        paddingHorizontal: 8,
        backgroundColor: 'transparent'
    },
    buttonText: {
        fontSize: 12,
    },
    buttonTextHide: {
        color: '#000'
    },
    buttonTextShow: {
        color: '#f22'
    },
    wrapper: {
        position: 'relative',
        width: '100%'

    }
});
