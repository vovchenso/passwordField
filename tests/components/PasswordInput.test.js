import React from 'react';
import PasswordInput from '../../src/components/PasswordInput/PasswordInput';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<PasswordInput />).toJSON();
  expect(rendered).toBeTruthy();
});

it('renders without props', () => {
    const rendered = renderer.create(<PasswordInput />).toJSON();
    expect(rendered).toMatchSnapshot();
});

it('renders with placeholder', () => {
    const rendered = renderer.create(
        <PasswordInput
            placeholder="TEST"
        />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
});

it('renders with change callback', () => {
    const rendered = renderer.create(
        <PasswordInput
            onChange={ () => {} }
        />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
});

it('renders with placeholder and callback', () => {
    const rendered = renderer.create(
        <PasswordInput
            placeholder="TEST"
            onChange={ () => {} }
        />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
});


