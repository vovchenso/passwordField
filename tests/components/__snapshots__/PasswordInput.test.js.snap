// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with change callback 1`] = `
<View
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder={undefined}
    secureTextEntry={true}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#a0a0a0",
          "height": 40,
          "padding": 5,
        },
        false,
        false,
      ]
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      SHOW
    </Text>
  </View>
</View>
`;

exports[`renders with placeholder 1`] = `
<View
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="TEST"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#a0a0a0",
          "height": 40,
          "padding": 5,
        },
        false,
        false,
      ]
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      SHOW
    </Text>
  </View>
</View>
`;

exports[`renders with placeholder and callback 1`] = `
<View
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="TEST"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#a0a0a0",
          "height": 40,
          "padding": 5,
        },
        false,
        false,
      ]
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      SHOW
    </Text>
  </View>
</View>
`;

exports[`renders without props 1`] = `
<View
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder={undefined}
    secureTextEntry={true}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#a0a0a0",
          "height": 40,
          "padding": 5,
        },
        false,
        false,
      ]
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      SHOW
    </Text>
  </View>
</View>
`;
